lib/ex_doc/formatter/html/templates.ex

```diff
@@ -157,7 +172,19 @@ defmodule ExDoc.Formatter.HTML.Templates do
   end
 
   def module_summary(module_node) do
-    [Types: module_node.typespecs] ++
+    sections =
+      case module_node.rendered_doc do
+        nil ->
+          []
+
+        moduledoc ->
+          Enum.map(extract_headers(moduledoc), &Map.put(&1, :type, :heading))
```

Comment:
This also is problematic. If there are multiple headings with the same text, this will always link to the first. I think I'll have to break out some of the code in `link_headings`

Reply:
I believe for pages, if we have duplicate titles, we add a indexed suffix.